lnrpc/routerrpc/router_server.go

```diff
@@ -627,12 +628,13 @@ func (s *Server) probePaymentRequest(ctx context.Context, paymentRequest string,
 // isLSP checks if the route hints indicate an LSP. An LSP is indicated with
 // true if the last node in each route hint has the same node id, false
 // otherwise.
-func isLSP(routeHints [][]zpay32.HopHint) bool {
+func isLSP(routeHints [][]zpay32.HopHint, routerBackend *RouterBackend) bool {
```

Comment:
why not make isLSP part of ther routerbackend `func (s *Server) isLSP` makes it way cleaner ?

Reply:
I think passing in a function in instead is good enough, also makes it easier to mock in a unit test

Reply:
done. I declared a new type `FetchChannelInfo` in the router_server.go which matches the function signature in router_backend.go.